Automattic/wp-calypso
# Replace ui.language.isRtl Redux state with HOC that depends on i18n-calypso

Several basic Calypso components like `Popover` depend on the `isRtl` Redux selector from `state/selectors/is-rtl` to render correctly in an RTL environment. Typically, tooltip positioning, inline CSS styles and arrow icons depend on the LTR/RTL flag.

Depending on Redux makes these components harder to extract to libraries and reuse. This PR removes the `isRtl` selector and the associated Redux state in favor of a `components/rtl` helper module that observes the state of the `i18n-calypso` object directly. There is a `useRtl` hook and a `withRtl` HOC.

**How to test:**
Enable RTL language like Hebrew and check that the `Chart` in Stats has all RTL features working:
- the X axis with labels is correctly rendered left-to-right
- the tooltip alignment is flipped when too near to the chart edge

<img width="635" alt="Screenshot 2019-10-01 at 12 00 38" src="https://user-images.githubusercontent.com/664258/65952978-4dc1d080-e443-11e9-8602-da8baa8bb5e2.png">
